### PR TITLE
Port `clojure-cheatsheet`

### LINKED ISSUE #9: Merge the clojure cheatsheet in helm-cider
https://github.com/clojure-emacs/clojure-cheatsheet is basically abandonware at this point and it just displays the Clojure cheatsheet using Helm. Seems reasonable to move the core logic to CIDER and expose it via two consecutive `completing-read` calls (or something like `browse-ns` type of interface) in CIDER itself and move the helm-related stuff here.

### PR DESCRIPTION
**Pull request by xiongtx**
Fix #9 

Depends on clojure-emacs/cider#2183.

### CODE REVIEW

File: helm-cider.el [hunk 1]
```diff
@@ -462,7 +464,7 @@ The old and new functions are those specified in
 
 ;;;###autoload
 (define-minor-mode helm-cider-mode
-  "Use Helm for CIDER."
+    "Use Helm for CIDER."
```
Comment (bbatsov) [inline helm-cider.el]: That looks wrong.

File: helm-cider-repl.el [hunk 2]
```diff
@@ -74,35 +75,37 @@ If not a positive integer, don't truncate candidate, show all."
...
-                             (save-excursion
-                               (forward-line helm-cider-repl-history-max-lines)
-                               (point)))
-                            "[...]"))
-                         c)
-           ;; Need to collect a cons b/c persistent action is executed on
-           ;; unpropertied string otherwise
-           ;; See: https://github.com/emacs-helm/helm/blob/v2.4.0/helm.el#L4999
-           else collect (cons c c)))
+  (cl-loop
```
Comment (bbatsov) [inline helm-cider-repl.el:86]: I think it's a bit uncommon to break `loop` in this manner, but I guess you did it to shorten the long lines.
Generally I dislike `loop` a lot and I think it usually produces somewhat hard to understand code (not to mention it's harder to debug macros).

Comment (xiongtx) [reply helm-cider-repl.el:86]: > I think it's a bit uncommon to break `loop` in this manner

I recently started [using `common-lisp-indent-function`](https://github.com/xiongtx/.emacs.d/commit/5c73d1f0c9fbe4fa23de5978d79d19a1bec80ae6) for Elisp to align forms for `cl-flet` etc. One consequence is that `cl-loop` forms no longer line up like:

```emacs-lisp
(cl-loop for x in foo
         for y in bar
         ...)
```

But like:

```emacs-lisp
(cl-loop for x in foo
   for y in bar
   ...)
```

Which is surprising b/c the former is [how CLTR formats `loop`](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node244.html#SECTION003060000000000000000). Nonetheless, I find moving the first clause onto a new line and keeping things lined up more pleasing.

> Generally I dislike `loop` a lot

I've come to take a sort of sadistic pleasure in using `loop`, perhaps influenced by Helm (which uses more `loop` than all of Emacs...I recall @Wilfred counted at some point :nerd_face:).

File: helm-cider.el (hunk 1 above)
Comment (xiongtx) [reply helm-cider.el]: Another consequence of using `common-lisp-indent-function` it seems...I guess I'll push the docstring up onto the same line as the `define-minor-mode`.

**Pull request merged**